llvm/lib/ProfileData/InstrProfReader.cpp

```diff
@@ -676,8 +679,14 @@ Error RawInstrProfReader<IntPtrT>::readRawCounts(
       // A value of zero signifies the block is covered.
       Record.Counts.push_back(*Ptr == 0 ? 1 : 0);
     } else {
-      const auto *CounterValue = reinterpret_cast<const uint64_t *>(Ptr);
-      Record.Counts.push_back(swap(*CounterValue));
+      uint64_t CounterValue = swap(*reinterpret_cast<const uint64_t *>(Ptr));
+      if (CounterValue > MaxCounterValue)
+        return error(instrprof_error::malformed,
+                     ("counter value " + Twine(CounterValue) +
+                      " is greater than " + Twine(MaxCounterValue))
+                         .str());
```

Comment:
Maybe hint to the user why this is an error condition?
```suggestion
        return error(instrprof_error::malformed,
                     ("counter value " + Twine(CounterValue) +
                      " is greater than " + Twine(MaxCounterValue) + " (unreasonably large)")
                         .str());
```

Reply:
I'd go even further and say something like "this is likely broken profile data"

Reply:
+1, "counter value 72057594037927936 is greater than 72057594037927935" isn't as helpful as "counter value 72057594037927936 suggests corrupted profile data"